crates/bevy_audio/src/audio_source.rs

```diff
@@ -80,22 +81,16 @@ impl AssetLoader for AudioLoader {
 /// This trait is implemented for [`AudioSource`].
 /// Check the example [`decodable`](https://github.com/bevyengine/bevy/blob/latest/examples/audio/decodable.rs) for how to implement this trait on a custom type.
 pub trait Decodable: Send + Sync + 'static {
-    /// The type of the audio samples.
-    /// Usually a [`u16`], [`i16`] or [`f32`], as those implement [`rodio::Sample`].
-    /// Other types can implement the [`rodio::Sample`] trait as well.
-    type DecoderItem: rodio::Sample + Send + Sync;
-
     /// The type of the iterator of the audio samples,
-    /// which iterates over samples of type [`Self::DecoderItem`].
+    /// which iterates over samples of type [`rodio::Sample`].
     /// Must be a [`rodio::Source`] so that it can provide information on the audio it is iterating over.
-    type Decoder: rodio::Source + Send + Iterator<Item = Self::DecoderItem>;
+    type Decoder: rodio::Source + Send + Iterator<Item = rodio::Sample>;
```

Comment:
`rodio::Source` has `Iterator<Item = rodio::Sample>` as a bound as far as I can see, should we drop this redundant bound or should we keep it?